### PR TITLE
fix: update group invite and removal notification tasks

### PR DESCRIPTION
**Pull request by katrinan029**
https://2u-internal.atlassian.net/browse/ENT-8857

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - Trigger the '[Upgrade one Python dependency](https://github.com/openedx/edx-platform/actions/workflows/upgrade-one-python-dependency.yml)' action against master in edx-platform with new version number to generate version bump PR
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/tasks.py [hunk 1]
```diff
@@ -255,16 +255,18 @@ def send_group_membership_invitation_notification(
...
-        recipients.append(braze_client_instance.create_recipient_no_external_id(pecu_email))
-    braze_client_instance.create_braze_alias(
-        [pecu_emails],
-        ENTERPRISE_BRAZE_ALIAS_LABEL,
-    )
-    for ecu in ecus:
-        recipients.append(braze_client_instance.create_recipient(
-            user_email=ecu['user_email'],
-            lms_user_id=ecu['user_id']))
+    if len(pecu_emails) > 0:
```
Comment (alex-sheehan-edx) [inline enterprise/tasks.py:258]: NIT not necessary but you can do `if pecu_emails:` and python should equate an empty array to False

File: enterprise/tasks.py [hunk 2]
```diff
@@ -255,16 +255,18 @@ def send_group_membership_invitation_notification(
...
+        for pecu_email in pecu_emails:
+            recipients.append(braze_client_instance.create_recipient_no_external_id(pecu_email))
+        braze_client_instance.create_braze_alias(
+            pecu_emails,
+            ENTERPRISE_BRAZE_ALIAS_LABEL,
+        )
+    if len(ecus) > 0:
```
Comment (alex-sheehan-edx) [inline enterprise/tasks.py:265]: I don't think this check is necessary- if `ecus` is empty, it will just not iterate through the loop

**Review by marlonkeating — CHANGES_REQUESTED** [review]
The implementation looks good as far as I can see, but could we add unit test(s) that would have caught the original issue?

**Review by alex-sheehan-edx — COMMENTED** [review]
Small nits that apply to both sections. LMK if you wanna chat about writing tests

**Review by marlonkeating — APPROVED** [review]

**Review by alex-sheehan-edx — APPROVED** [review]

**Pull request merged**